matplotcheck/base.py

```diff
@@ -183,19 +197,29 @@ def assert_caption_contains(self, strings_exp):
         """Asserts that Axes ax contains strings as expected in strings_exp.
         strings_exp is a list of lists. Each internal list is a list of
         strings where at least one string must be in the caption, barring
-        capitalization. Once a string is found, it is removed from the
-        caption, therefore, order does matter. This is to enforce no overlap
-        in found strings.
+        capitalization. For example, ``assert_caption_contains([['A'], ['B'], ['C']])``
+        checks that all of ``'A'``, ``'B'``, and ``'C'`` exist in the caption.
+        Alternatively, ``assert_caption_contains([['A', 'B', 'C']])`` checks
+        that any one of ``'A'``, ``'B'``, and ``'C'`` exist in the caption.
 
         Parameters
         ----------
         strings_exp: list of lists.
-            Each internal list is a list of strings
-            where at least one string must be
-            found in the caption. Input strings must be lower case, as we are
-            not testing for capitalization
-            if None: assert caption does not exist
-            if empty list: asserts caption exists and not an empty string
+            list of lists in which each internal list is a list of strings.
+            If ``strings_exp = None``, assertion passes.
+            If ``strings_exp == []``, assertion passes.
+
+        Returns
+        -------
+        None :
+            Nothing if caption contains strings matching `strings_exp`, otherwise throws ``AssertionError``
+
+        Notes
+        -----
+            This function enforces that no found strings can overlap.
```

Comment:
oh that is interesting. so that could really impact how we test things. i can't think of a reason why we'd want to REMOVE a string if found. i think we'd just want to check if it's there or not. I suggest you create a separate issue for this and we can refactor this method to ensure it's not removing parts of the string once it finds it.

Reply:
I don't think it removes the found string from the actual caption. I think it stores the caption text in a local variable and removes the found string from that local variable.

In short, the removal doesn't change the actual caption, it just prevents two of the expected strings from matching on the same portion of the caption.

Are things okay as they are if the actual caption is not being changed? Or do you want to allow two expected strings to match on the same portion of the caption?

Reply:
i think we want to allow two expected strings to match in the same portion of the caption IF it is an OR. let me think about this a bit more. it's just not how i had thought that method was operating. i am going to copy this discussion into a new issue so i can still merge this pr.